### ISSUE TITLE
Implicit dependencies are ignored when deciding wether to release or not

### ISSUE THREAD

**Issue by ianldgs** [jscutlery/semver#477]
NX adds this by default on nx.json:

```
"implicitDependencies": {
    "package.json": {
      "dependencies": "*",
      "devDependencies": "*"
    },
    ".eslintrc.json": "*"
  },
```

So that if you update your external dependencies, everything has to be rebuilt.
People can also configure their own implicit dependencies.

The `@jscutlery/semver:version` executor does not take those implicit dependencies into account and outputs "⏹ Nothing changed since last release." when I bump my dependencies.

While I agree that NX could be smarter regarding external dependencies, it is important that if a library's external dependency is updated, it is re-linted, tested, built and gets a new release.
Otherwise, any bug fixes of our dependencies are not released in our libraries.

Ideally, I'd like to bump at least a patch version whenever any of my libraries are affected by anything. Thoughts? 

Maybe we can add a new option for this, if you're not willing to have this behaviour by default (I could open a PR).
Another option would be to track the implicit dependencies, but it would be more work.

**Comment by lamanabie** [comment]
@ianldgs Ideally, I'd like to bump at least a patch version whenever any of my libraries are affected by anything. Thoughts? 
u can use `--releaseAs=patch`, to force bump even if there are no changes

**Comment by ianldgs** [comment]
Hey @lamanabie! Sure, but then you force EVERY release you make to be a `patch`.
Even if you have a commit that made a breaking change, you'd only get a  0.0.1 bump.
Wouldn't be semver anymore :)

**Comment by ianldgs** [comment]
@edbzn what would be your preferred implementation here?
I can make the change.
This is blocking me from updating to NX 13.8+ and 13.10 was just released.

**Comment by edbzn** [comment]
@ianldgs Yeah I think it's a good addition, we should do the calculation if --trackDeps option is passed.

**Comment by ianldgs** [comment]
But would you be ok with an option, for example, `releaseAtLeast: 'patch'`? 

I don't think I could commit to making the change to look at all of the implicit dependencies.
And TBH, if people are running this library with `affected`, NX would in theory already be doing this lookup of the implicit dependencies for us and only calling the `version` target if
